README.md

```diff
@@ -8,92 +8,196 @@
+a column, use the syntax `cols(varname)`. 
 
-df = DataFrame(x = 1:3, y = [2, 1, 2])
-x = [2, 1, 0]
+Use `passmissing` from [Missings.jl](https://github.com/JuliaData/Missings.jl) to 
+propagate `missing` values more easily. 
 
-@with(df, :y .+ 1)
-@with(df, :x + x)  # the two x's are different
+Details below: 
 
-x = @with df begin
-    res = 0.0
-    for i in 1:length(:x)
-        res += :x[i] * :y[i]
-    end
-    res
-end
+# Details
 
-@with(df, df[:x .> 1, ^(:y)]) # The ^ means leave the :y alone
+## `@select`
+
+Column selections and transformations. Only newly created columns are kept. 
+Operates on both a `DataFrame` and a `GroupedDataFrame`. 
 
+```julia
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+@select(df, :x, :y)
+@select(df, x2 = 2 * :x, :y)
+@select(gd, x2 = 2 .* :y .* first(:y))
 ```
 
-`@with` is the fundamental macro used by the other metaprogramming
-utilities.
+!!! note 
+    
+    Versions of DataFrames.jl `0.19` and above support the operators `Between`, `All`,
+    and `Not` when selecting and transforming columns. DataFramesMeta.jl does not currently
+    support this syntax. 
+
+!!! note
+    To refer to `Symbol`s without aliasing the column in a DataFame, use `^`. 
+
+    ```
+    df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+    @select(df, :x2 = :x, :x3 = ^(:x))
+    ```
+
+    This rule applies to all DataFramesMeta macros.
 
-`@with` creates a function, so scope within `@with` is a local scope.
-Variables in the parent can be read. Writing to variables in the parent scope
-differs depending on the type of scope of the parent. If the parent scope is a
-global scope, then a variable cannot be assigned without using the `global` keyword.
-If the parent scope is a local scope (inside a function or let block for example),
-the `global` keyword is not needed to assign to that parent scope.
+## `@transform`
+
+Add additional columns based on keyword arguments. Operates on both a 
+`DataFrame` and a `GroupedDataFrame`. 
+
+```julia
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+@transform(df, :x, :y)
+@transform(df, x2 = 2 * :x, :y)
+@transform(gd, x2 = 2 .* :y .* first(:y))
+```
 
 ## `@where`
 
-Select row subsets.
+Select row subsets. Operates on both a `DataFrame` and a `GroupedDataFrame`. 
 
 ```julia
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+x = 1;
 @where(df, :x .> 1)
 @where(df, :x .> x)
-@where(df, :x .> x, :y .== 3)  # the two expressions are "and-ed"
+@where(df, :x .> x, :y .< 102)  # the two expressions are "and-ed"
+@where(gd, :x .> mean(:x))
 ```
 
-## `@select`
+## `@combine`
+
+Summarize or collapse, a grouped data frame by perforing transformations at the group level and 
+collecting the result into a single data frame. Also works on a `DataFrame`, which 
+acts like a `GroupedDataFrame` with one group. 
```

Comment:
Grouped operations are also supported by `@select` and `@transform`: better explain what happens with `GroupedDataFrame` in general before presenting separate macros.
Also:
```suggestion
Summarize or collapse a grouped data frame by performing transformations at the group level and 
collecting the result into a single data frame. Also works on a `DataFrame`, which 
acts like a `GroupedDataFrame` with one group. 
```

Reply:
Added a one sentence explainer